File: algorithm/src/programmers/visitLength.java
package programmers;
class Solution42 {
    public int solution(String dirs) {
        int answer = 0, check = 0;
        int[][] visit = new int[11][11];
        int x = 6, y = 5;
        visit[x][y] = 1;
        for(int k=0; k<dirs.length(); k++){
            if(dirs.charAt(k)=='U') {
                if(y==0) continue;
                else visit[x][--y] += 1;
            }
            else if(dirs.charAt(k)=='D') {
                if(y==10) continue;
                else visit[x][++y] += 1;
            }
            else if(dirs.charAt(k)=='R') {
                if(x==10) continue;
                else visit[++x][y] += 1;
            }
            else {
                if(x==1) continue;
                else visit[--x][y] += 1;
            }
            if(visit[x][y] == 2) check++;
            if(visit[x][y] == 1 || check == 1) answer++;
        }
        return answer;
    }
}
public class visitLength {
	public static void main(String[] args) {
		Solution42 sol = new Solution42();
		String dirs = "ULURRDLLU";
		System.out.println("방문 길이 : "+sol.solution(dirs));
	}
}

File: algorithm/src/programmers/visitLength.py
def solution(dirs):
    answer = 0
    direct = {'U':[0,1], 'D':[0,-1], 'L':[-1,0], 'R':[1,0]}
    V = []
    nx,ny = 0,0
    
    for d in dirs :
        cx,cy = nx,ny
        nx += direct[d][0]
        ny += direct[d][1]
        
        if nx<-5 or nx>5 or ny<-5 or ny>5 :
            nx -= direct[d][0]
            ny -= direct[d][1]
            continue
        if [nx,ny,cx,cy] in V :
            continue
        if [cx,cy,nx,ny] not in V :
            answer += 1
        
        V.append([cx,cy,nx,ny])
    
    return answer

dirs = "ULURRDLLU"
print("-5<=x<=5, -5<=y<=5를 만족하는 좌표평면이 있습니다.")
print("좌표평면의 [0,0]부터 시작해 \"",dirs,"\" 명령대로 움직였을 때 처음 방문하는 좌표의 수 : ",solution(dirs))